remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/util/LazySupplier.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.query.remote.impl.util;
+
+import java.util.function.Supplier;
+
+/**
+ * @since 9.2
+ */
+public class LazySupplier<R> implements Supplier<R> {
```

Comment:
I think this should be called `MemoizedSupplier`.

Reply:
I was trying to copy Scala's ```lazy``` behavior, so I'll rename it to ```LazyRef``` instead